From 3b1aec9b8ff43f81cf7df9be92d0d383d759792e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Fri, 28 Jun 2024 07:52:59 +0200
Subject: [PATCH] Update class method deprecation warnings for Python 3.13

Update the version range for modified deprecation warnings that was
introduced in efb3e60623e1dda88c2725a93223d290924e8666, since Python
3.13 reverted the change originally introduced in 3.9 and is back to
the old messages.  This fixes tests with Python 3.13.

See also https://github.com/GrahamDumpleton/wrapt/pull/260.
---
 tests/test_deprecated.py | 2 +-
 tests/test_sphinx.py     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/test_deprecated.py b/tests/test_deprecated.py
index 0e467ae..c39229b 100644
--- a/tests/test_deprecated.py
+++ b/tests/test_deprecated.py
@@ -191,7 +191,7 @@ def test_classic_deprecated_class_method__warns(classic_deprecated_class_method)
     assert len(warns) == 1
     warn = warns[0]
     assert issubclass(warn.category, DeprecationWarning)
-    if sys.version_info >= (3, 9):
+    if (3, 9) <= sys.version_info < (3, 13):
         assert "deprecated class method" in str(warn.message)
     else:
         assert "deprecated function (or staticmethod)" in str(warn.message)
diff --git a/tests/test_sphinx.py b/tests/test_sphinx.py
index a1d7753..8cf8e95 100644
--- a/tests/test_sphinx.py
+++ b/tests/test_sphinx.py
@@ -301,7 +301,7 @@ def test_sphinx_deprecated_class_method__warns(sphinx_deprecated_class_method):
     assert len(warns) == 1
     warn = warns[0]
     assert issubclass(warn.category, DeprecationWarning)
-    if sys.version_info >= (3, 9):
+    if (3, 9) <= sys.version_info < (3, 13):
         assert "deprecated class method" in str(warn.message)
     else:
         assert "deprecated function (or staticmethod)" in str(warn.message)
